From d8c8677f6c1aa26dd7858037b77bfc3314d2e62e Mon Sep 17 00:00:00 2001
From: arunasank <aruna@mapbox.com>
Date: Sun, 17 Jul 2016 16:51:29 +0000
Subject: [PATCH 1/2] condition

---
 identify-highways/map.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/identify-highways/map.js b/identify-highways/map.js
index 7a0a4ae..ae0c0dd 100644
--- a/identify-highways/map.js
+++ b/identify-highways/map.js
@@ -14,12 +14,14 @@ module.exports = function (data, tile, writeData, done) {
             totalFatalities += accident.properties.fatalities;
         });
     }
+    data.highways.highways.features.forEach(function (highwayFeature) {
+        if(highwayFeature.properties.highway == 'primary' || highwayFeature.properties.highway == 'primary_link' || highwayFeature.properties.highway == 'secondary' || highwayFeature.properties.highway == 'secondary_link' || highwayFeature.properties.highway == 'tertiary' || highwayFeature.properties.highway == 'tertiary_link' || highwayFeature.properties.highway == 'motorway' || highwayFeature.properties.highway == 'motorway_link') {
 
-    data.highways.highway.features.forEach(function (highwayFeature) {
         highwayFeature.properties.totalAccidents = totalAccidents;
         highwayFeature.properties.totalFatalities = totalFatalities;
+        writeData(highwayFeature);
+        }
     });
 
-    writeData(data.highways.highway.features);
     done(null, null);
 };
-- 
1.9.1

